test-vectors/schema.json

```diff
@@ -0,0 +1,156 @@
+        },
+        "gen": {
+          "title": "generation metadata",
+          "description": "metadata about how this test vector was generated",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "source": {
+              "type": "string",
+              "examples": [
+                "lotus",
+                "dsl"
+              ]
+            },
+            "version": {
+              "type": "string",
+              "examples": [
+                "0.4.1+git.27d74337+api0.8.1"
+              ]
+            }
+          }
+        }
+      }
+    },
+    "preconditions": {
+      "title": "execution preconditions",
+      "description": "preconditions that need to be applied and satisfied before this test vector can be executed",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "state_tree": {
+          "title": "state tree to seed",
+          "description": "state tree to seed before applying this test vector; mapping of actor addresses => serialized state",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/hex"
+          },
+          "examples": [
+            {
+              "t01": "0x0123456789abcdef",
+              "t02": "0x0123456789abcdef",
+              "t03": "0x0123456789abcdef"
+            }
+          ]
+        }
+      }
+    },
+    "postconditions": {
+      "title": "execution preconditions",
+      "description": "postconditions that need to be satisfied after execution for this test vector to pass",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "state_tree": {
+          "title": "state tree postconditions",
+          "description": "state tree postconditions that must be true for this test vector to pass",
+          "type": "object",
+          "properties": {
+            "root_cid": {
+              "title": "the root CID of the state tree",
+              "type": "string"
+            },
+            "car_bytes": {
+              "title": "the hex-encoded CAR containing the full state tree",
+              "description": "the hex-encoded CAR containing the full state tree, for debugging/diffing purposes",
+              "$ref": "#/definitions/hex"
+            }
+          }
+        }
+      }
+    },
+    "apply_messages": {
+      "title": "list of messages to apply, hex-encoded",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/hex"
+      }
+    }
+  },
+  "required": [
+    "class"
+  ],
+  "properties": {
+    "class": {
+      "title": "test vector class",
+      "description": "test vector class; depending on the value, the apply_* property to provide (and its schema) will vary; the relevant apply property is apply_[class]",
+      "type": "string",
+      "enum": [
+        "messages",
+        "block",
+        "tipset",
+        "chain"
+      ]
+    },
```

Comment:
I understand messages can transition the state. I'm not clear on what a test vector for the other classes listed here might look like - are they just examples at this point?

How would we express a test vector that applies messages, blocks and then some more messages and then asserts on the state? ...is this too high level?